From cb6b54e5b66d9efb69f4c1515673cb74687a61c5 Mon Sep 17 00:00:00 2001
From: Leo Wang <leowang@nexcom.com.tw>
Date: Wed, 26 Mar 2025 09:02:54 +0800
Subject: [PATCH] nvpower: add nvpmodel symlink funciton.

---
 etc/systemd/nvpower.sh | 27 +++++++++++++++++++++++++--
 1 file changed, 25 insertions(+), 2 deletions(-)

diff --git a/etc/systemd/nvpower.sh b/etc/systemd/nvpower.sh
index 3afc042..b536a6f 100755
--- a/etc/systemd/nvpower.sh
+++ b/etc/systemd/nvpower.sh
@@ -140,6 +140,25 @@ function set_power_state_perm()
 	fi
 }
 
+function create_nvpmodel_symlink()
+{
+	conf_file=""
+	if [ "${SOCFAMILY}" = "tegra234" ]; then
+		if [ "${machine}" = "p3701-0004" ] || \
+		   [ "${machine}" = "p3701-0005" ]; then
+			conf_file="/etc/nvpmodel/nvpmodel_p3701_0004.conf"	
+		fi
+	fi
+	
+	if [ "${conf_file}" != "" ]; then
+		if [ -e "${conf_file}" ]; then
+			ln -sf "${conf_file}" /etc/nvpmodel.conf
+		else
+			echo "${SCRIPT_NAME} - WARNING: file ${conf_file} not found!"
+		fi
+	fi
+}
+
 # CPU hotplug helper function that turns on/off the CPU cores in the specified range
 # * Parameters:
 # * ${1}: desired online status (0 for offline, 1 for online)
@@ -395,9 +414,12 @@ function config_hwmon()
 		then
 			case "${1}" in
 			ina3221)
-				if [ "$(< ${dir}/in1_label)" != "${4}" ]
-				then continue
+				if [ ! -f "${dir}/in1_label" ]; then
+					continue
 				fi
+				if [ "$(< ${dir}/in1_label)" != "${4}" ]; then
+					continue
+					fi
 				;;
 			*)
 				echo "unsupported condition"
@@ -477,6 +499,7 @@ SOCFAMILY=""
 set_socfamily
 set_safety_flag
 set_power_state_perm
+create_nvpmodel_symlink
 cpu_hotplug
 set_cpufreq_governor
 set_cpu_floor_freq
-- 
2.25.1

